src/main/java/io/passport/server/service/ModelDeploymentService.java

```diff
@@ -0,0 +1,119 @@
+package io.passport.server.service;
+
+import io.passport.server.model.ModelDeployment;
+import io.passport.server.repository.ModelDeploymentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Service class for model deployment management.
+ */
+@Service
+public class ModelDeploymentService {
+
+    /**
+     * ModelDeployment repo access for database management.
+     */
+    private final ModelDeploymentRepository modelDeploymentRepository;
+
+    /**
+     * Deployment Environment service access.
+     */
+    private final DeploymentEnvironmentService deploymentEnvironmentService;
+
+
+    @Autowired
+    public ModelDeploymentService(ModelDeploymentRepository modelDeploymentRepository, DeploymentEnvironmentService deploymentEnvironmentService) {
+        this.modelDeploymentRepository = modelDeploymentRepository;
+        this.deploymentEnvironmentService = deploymentEnvironmentService;
+    }
+
+    /**
+     * Return all model deployments
+     * @return
+     */
+    public List<ModelDeployment> getAllModelDeployments() {
+        return modelDeploymentRepository.findAll();
+    }
+
+
+
+    /**
+     * Find a model deployment by deploymentId
+     * @param deploymentId ID of the model deployment
+     * @return
+     */
+    public Optional<ModelDeployment> findModelDeploymentByDeploymentId(Long deploymentId) {
+        return modelDeploymentRepository.findById(deploymentId);
+    }
+
+
+    /**
+     * Save a model deployment
+     * @param modelDeployment model deployment to be saved
+     * @return
+     */
+    public ModelDeployment saveModelDeployment(ModelDeployment modelDeployment) {
+        // Set the creation and last update timestamps
+        LocalDateTime now = LocalDateTime.now();
```

Comment:
Let's use Instant class for time class. Please update all time class usages accordingly.